l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -302,8 +302,15 @@ library Decoder {
 
       // Create the leaf to contain commitments (2 * COMMITMENTS_PER_TX * 020) + nullifiers (2 * NULLIFIERS_PER_TX * 0x20)
       // + new public data writes (8 * 0x40) + contract deployments (2 * 0x60) + logs hashes (2 * 4 * 0x20)
+      ArrayLengths memory constantsLengthsPerBase;
+      constantsLengthsPerBase.commitmentCount = 2 * Constants.COMMITMENTS_PER_TX;
+      constantsLengthsPerBase.nullifierCount = 2 * Constants.NULLIFIERS_PER_TX;
+      constantsLengthsPerBase.dataWritesCount = 2 * Constants.PUBLIC_DATA_WRITES_PER_TX;
+      constantsLengthsPerBase.l2ToL1MsgsCount = 2 * Constants.L2_TO_L1_MSGS_PER_TX;
+      constantsLengthsPerBase.contractCount = 2 * Constants.CONTRACTS_PER_TX;
+
       vars.baseLeaf =
-      new bytes(2 * Constants.COMMITMENTS_PER_TX * 0x20 + 2 * Constants.NULLIFIERS_PER_TX * 0x20 + 2 * Constants.PUBLIC_DATA_WRITES_PER_TX * 0x40 + 2 * Constants.CONTRACTS_PER_TX * 0x60 + 2 * 4 * 0x20);
+      new bytes(constantsLengthsPerBase.commitmentCount * 0x20 + constantsLengthsPerBase.nullifierCount * 0x20 + constantsLengthsPerBase.dataWritesCount * 0x40 + constantsLengthsPerBase.contractCount * 0x60 + 2 * 4 * 0x20);
```

Comment:
Can you add the note on 2 * 4 * 0x20, add to the constants to have the logs accounted for.